ooniprobe: Update to v3.14.2

jamesmacwhite opened the pull request:
Signed-off-by: James White <james@jmwhite.co.uk>

Maintainer: @ja-pa
Compile tested: mvebu/Linksys WRT3200ACM 21.02 and bcm53xx/generic master
Run tested: mvebu/Linksys WRT3200ACM OpenWrt 21.02

Description:

Since the Go 1.18 update ooniprobe fails to build for master/snapshot

https://downloads.openwrt.org/snapshots/faillogs/x86_64/packages/ooniprobe/compile.txt

The version of the package is quite old. Bumping it to 3.14.2 allows the package to compile without the error, as the Go 1.18 incompatibility has been handled upstream. https://github.com/openwrt/packages/pull/13642#issuecomment-707394122

The latest version can be backported down to 21.02 to fix this issue: https://github.com/openwrt/packages/issues/18219

jamesmacwhite commented:
Run tested on 21.02

```
root@linksys-wrt3200acm:~# ooniprobe version
3.14.1

root@linksys-wrt3200acm:~# apitool --help
Usage of apitool:
  -input string
        Input of the measurement
  -mode string
        One of: check, meta, raw
  -report-id string
        Report ID of the measurement
  -v    Enable verbose mode

root@linksys-wrt3200acm:~# miniooni --version
3.14.1
```

jefferyto commented:
Would you mind also removing `GO_PKG_TAGS:=PSIPHON_DISABLE_QUIC`? It should no longer be necessary (see https://github.com/openwrt/packages/pull/13642#issuecomment-707394122).

jamesmacwhite commented:
@jefferyto Sure, updated the Makefile with that removed now.

jamesmacwhite commented:
@jefferyto Removing that seemed to have regressed the build to fail to compile with the CIs. Are we sure isn't still not needed?

jamesmacwhite commented:
```
Building targets
github.com/ooni/psiphon/tunnel-core/oovendor/quic-go/internal/qtls
# github.com/ooni/psiphon/tunnel-core/oovendor/quic-go/internal/qtls
../../../../../dl/go-mod-cache/github.com/ooni/psiphon/tunnel-core@v0.0.0-20220121102537-7fc0bcd97c5f/oovendor/quic-go/internal/qtls/go118.go:5:13: cannot use "quic-go doesn't build on Go 1.18 yet." (untyped string constant) as int value in variable declaration

make[2]: *** [Makefile:49: /home/build/openwrt/build_dir/target-arm_cortex-a9+vfpv3-d16_musl_eabi/probe-cli-3.14.1/.built] Error 2
make[2]: Leaving directory '/home/runner/work/packages/packages/net/ooniprobe'
time: package/feeds/packages_ci/ooniprobe/compile#27.49#5.26#17.25
    ERROR: package/feeds/packages_ci/ooniprobe failed to build.
make[1]: *** [package/Makefile:116: package/feeds/packages_ci/ooniprobe/compile] Error 1
make[1]: Leaving directory '/home/build/openwrt'
make: *** [/home/build/openwrt/include/toplevel.mk:223: package/ooniprobe/compile] Error 2
Error: Process completed with exit code 2.
```

jefferyto commented:
I've opened https://github.com/ooni/probe/issues/2066 but I don't know how quickly it will be resolved. I don't mind if you want to keep `GO_PKG_TAGS:=PSIPHON_DISABLE_QUIC` to get this merged. (Apologies for the trouble.)

jamesmacwhite commented:
@jefferyto No trouble at all, I think ironically, it has been masking the Go 1.18 issue, which otherwise would have been there, but it does build when disabling QUIC and skips the affected dependency currently, so maybe we should keep it, even though it's original purpose wasn't for this.

The main urgency to get this merged is to then backport it down to 21.02, which currently has a broken ooniprobe binary being provided in the packages feed.

jamesmacwhite commented:
@BKPepe This has since been run tested as well.

jamesmacwhite commented:
@jefferyto Thanks for the notification. I've updated to 3.14.2 and removed the build flag disabling QUIC, let's see if the build CI is happier with it.

jamesmacwhite commented:
Good to go, run tested on snapshot and 21.02.

jefferyto commented:
Thanks - would you mind squashing the commits, since we're skipping v3.14.1?

jamesmacwhite commented:
@jefferyto Ah sorry about that, thought I'd rebased it already. Sorted now.

jefferyto commented:
Thanks!
